packages/langium/src/lsp/language-server.ts

```diff
@@ -215,30 +218,39 @@ export function startLanguageServer(services: LangiumSharedServices): void {
 }
 
 export function addDocumentsHandler(connection: Connection, services: LangiumSharedServices): void {
-    const documentBuilder = services.workspace.DocumentBuilder;
-    const mutex = services.workspace.MutexLock;
-
-    function onDidChange(changed: URI[], deleted: URI[]): void {
-        mutex.lock(token => documentBuilder.update(changed, deleted, token));
+    const handler = services.lsp.DocumentUpdateHandler;
+    const documents = services.workspace.TextDocuments;
+    if (handler.didChangeContent) {
```

Comment:
`didChangeContent` and ` didChangeWatchedFiles` are mandatory according you definitions in `DocumentUpdateHandler`, right? So these guards here and below could be dropped.